pkg/controllers/user/logging/utils/embedded.go

```diff
@@ -282,30 +287,38 @@ func updateEmbeddedQuota(dep rv1beta2.DeploymentInterface, obj *v3.ClusterLoggin
 	return nil
 }
 
-func getEndpoint(deploymentLister rv1beta2.DeploymentLister, endpointLister rv1.EndpointsLister, serviceLister rv1.ServiceLister, nodeLister v3.NodeLister, k8sNodeLister rv1.NodeLister, clusterName, serviceName string) (esEndpoint string, err error) {
-	deploymen, err := deploymentLister.Get(loggingconfig.LoggingNamespace, loggingconfig.EmbeddedESName)
-	for _, cond := range deploymen.Status.Conditions {
-		if cond.Status == v1.ConditionFalse {
-			return "", fmt.Errorf("deployment %s status %s is %s, reason: %s, message: %s", deploymen.Name, cond.Type, cond.Status, cond.Reason, cond.Message)
-		}
+func getEndpoint(podLister rv1.PodLister, serviceLister rv1.ServiceLister, nodeLister v3.NodeLister, k8sNodeLister rv1.NodeLister, obj *v3.ClusterLogging, clusterName, serviceName string, checkResourceQuota bool) (endpoint, waitingMsg string, err error) {
+	selector := labels.NewSelector()
+	requirement, err := labels.NewRequirement(loggingconfig.LabelK8sApp, selection.Equals, []string{serviceName})
+	if err != nil {
+		return "", "", err
 	}
 
-	endpoint, err := endpointLister.Get(loggingconfig.LoggingNamespace, serviceName)
+	pods, err := podLister.List(loggingconfig.LoggingNamespace, selector.Add(*requirement))
 	if err != nil {
-		return "", err
+		return "", "", err
+	}
+	if len(pods) == 0 {
+		return "", fmt.Sprintf("waiting for %s pod deploy", serviceName), nil
 	}
+	pod := pods[0]
 
-	if len(endpoint.Subsets) == 0 || len(endpoint.Subsets[0].Addresses) == 0 {
-		return "", fmt.Errorf("get %s endpoint subsets failed", serviceName)
+	if checkResourceQuota && !isLatestResourceQuota(pod, obj) {
+		return "", fmt.Sprintf("waiting for %s reconcile", serviceName), nil
```

Comment:
Nit: serviceName might not be unique cross namespaces, so it would probably make sense to add the namespace to the error message. You can format name as `namespace/name` - standard k8s way